### PR TITLE
Add Searching by Object ID (UUID)

### PR DESCRIPTION
**Pull request by zackt25**
# Closes: #923
# What's Changed

Added the ability to the existing search functions to be able to query objects using UUID. Added "| Q(id__iexact=value)" to all search functions.

Examples:

### Searching for Circuit 7d96148d-9069-44e6-a71c-fa787572ae3c
image

**Before:**
image

**After:**
image

### Searching for Device b5a70402-82a9-4869-b462-5c8661bc8de2
image

**Before:**
image

**After:**
image

### Searching for Site 38f5fc49-b4d5-43b6-9963-e4cb725a5f89
image

**Before:**
image

**After:**
image

# TODO

- [x] Explanation of Change(s)
- [x] Attached Screenshots, Payload Example
- [x] Documentation Updates (when adding/changing features)

### CODE REVIEW

File: nautobot/extras/filters.py [hunk 1]
```diff
@@ -976,7 +997,9 @@ def search(self, queryset, name, value):
         if not value.strip():
             return queryset
         return queryset.filter(
-            Q(name__icontains=value) | Q(slug__icontains=value) | Q(content_types__model__icontains=value)
+            Q(name__icontains=value)
+            | Q(slug__icontains=value)
+            | Q(content_types__model__icontains=value) | Q(id__iexact=value)
```
Comment (bryanculver) [inline nautobot/extras/filters.py:1002]: It may not be over the line length but we should be consistent to move `| Q(id__iexact=value)` to it's own line.

Comment (zackt25) [reply nautobot/extras/filters.py]: I thought I caught it, but it should be fixed with new push

**Comment by bryanculver** [comment]
Thanks @zackt25!

Mind adding some test cases to test this search, something like this: https://github.com/nautobot/nautobot/blob/develop/nautobot/ipam/tests/test_filters.py#L535-L545

**Comment by jathanism** [comment]
One thing to note is that `id__exact=value` is the same as `id=value`, so that extra lookup is redundant for that.

Another thing is perhaps we should revisit this in the context of #1752 and perhaps just always add `id=value` to any instance of `SearchFilter` automatically?

**Comment by zackt25** [comment]
> One thing to note is that `id__exact=value` is the same as `id=value`, so that extra lookup is redundant for that.
> 
> Another thing is perhaps we should revisit this in the context of #1752 and perhaps just always add `id=value` to any instance of `SearchFilter` automatically?

I think that would be a good idea. I see that you remove all the search functions so that would mean that I don't think we would need this PR and the issue could be moved over to #1752 as the fix.

**Comment by jathanism** [comment]
> > One thing to note is that `id__exact=value` is the same as `id=value`, so that extra lookup is redundant for that.
> > Another thing is perhaps we should revisit this in the context of #1752 and perhaps just always add `id=value` to any instance of `SearchFilter` automatically?
> 
> @jathanism I think that would be a good idea. I see that you remove all the search functions so that would mean that I don't think we would need this PR and the issue could be moved over to #1752 as the fix.
> 
> Should I just close this PR?

Yeah let's just close this one. Thank you!

**Pull request closed without merging**